worldedit-core/src/main/java/com/sk89q/worldedit/command/argument/OffsetConverter.java

```diff
@@ -62,15 +69,66 @@ public Component describeAcceptableArguments() {
 
     @Override
     public List<String> getSuggestions(String input, InjectedValueAccess context) {
+        if (input.startsWith("^") && context.injectedValue(Key.of(Actor.class))
+            .filter(actor -> actor instanceof Locatable).isPresent()) {
+            return vectorConverter.getSuggestions(input.substring(1), context).stream()
+                .map(s -> "^" + s)
+                .collect(Collectors.toList());
+        }
         return ImmutableList.copyOf(Iterables.concat(
             directionVectorConverter.getSuggestions(input, context),
             vectorConverter.getSuggestions(input, context)
         ));
     }
 
+    private BlockVector3 rotateToRelative(Location location, BlockVector3 relativeOffset) {
+        float pitch = location.getPitch();
+        float yaw = location.getYaw();
+
+        // This math was borrowed from the MC codebase, with some changes made for accuracy
+        double f = MathUtils.dCos(yaw + 90.0);
+        double g = MathUtils.dSin(yaw + 90.0);
+        double j = MathUtils.dCos(-pitch + 90.0);
+        double k = MathUtils.dSin(-pitch + 90.0);
+
+        Vector3 m1 = location.getDirection();
+        Vector3 m2 = Vector3.at(f * j, k, g * j);
+        Vector3 m3 = m1.cross(m2).multiply(-1.0D);
+
+        // Create an affine transform of the columns (col4 is empty due to no translation)
+        AffineTransform transform = new AffineTransform(
+            m1.getX(), m2.getX(), m3.getX(), 0,
+            m1.getY(), m2.getY(), m3.getY(), 0,
+            m1.getZ(), m2.getZ(), m3.getZ(), 0
+        );
+
+        return transform
+            .apply(relativeOffset.toVector3())
+            // This *MUST* be rounded before converting to block points
+            .round().toBlockPoint();
+    }
+
     @Override
     public ConversionResult<BlockVector3> convert(String input, InjectedValueAccess context) {
-        return directionVectorConverter.convert(input, context)
-            .orElse(vectorConverter.convert(input, context));
+        if (input.startsWith("^")) {
+            // Looking at a relative vector.
+            Actor actor = context.injectedValue(Key.of(Actor.class))
+                .orElseThrow(() -> new IllegalStateException("An actor is required to use relative offsets"));
```

Comment:
oh yea, you're uh, not supposed to throw here. Return a `FailedConversion.from(ex)` instead, or just wrap the whole thing in a `try-catch` :P